kimchi/src/verifier.rs

```diff
@@ -805,22 +817,36 @@ where
 
     //~ 1. Ensure that all the proof's verifier index have a URS of the same length. (TODO: do they have to be the same URS though? should we check for that?)
     // TODO: Account for the different SRS lengths
-    let srs = &proofs[0].0.srs();
-    for (index, _) in proofs.iter() {
-        if index.srs().g.len() != srs.g.len() {
+    let srs = &proofs[0].verifier_index.srs();
+    for BatchedProof {
+        verifier_index,
+        proof: _,
+        public_input: _,
+    } in proofs.iter()
```

Comment:
nit: you can simply write `for BatchedProof { verifier_index, .. } in &proofs` here

Reply:
Done